compendium/LogServiceImpl/src/LogServiceImpl.cpp

```diff
@@ -6,93 +6,187 @@
 #include "cppmicroservices/ServiceReference.h"
 
 #include "LogServiceImpl.hpp"
+#include "LoggerFactoryImpl.hpp"
 
-namespace cppmicroservices
-{
-    namespace logservice
-    {
-        std::string
-        GetExceptionMessage(std::exception_ptr const& ex)
+namespace cppmicroservices::logservice
+{     
+        LogServiceImpl::LogServiceImpl(std::string const& loggerName)
         {
-            std::string message = "\nException logged: ";
-            if (ex)
+            logger = getLogger(loggerName);    
+	}
+
+        void
+        LogServiceImpl::Log(SeverityLevel level, std::string const& message)
+        {
+            switch (level)
```

Comment:
Lots of repeated code below. Consider an early exit at the top of the function:

auto currLogger = std::atomic_load(&logger);
if (!currLogger) {
    return;
}

Then all of the cases in the switch become one line.